test/test.js

```diff
@@ -822,5 +822,41 @@ exports.shuffleArray = shuffleArray_1;
 `
 			);
 		});
+
+		it.only('normalizes paths used in the named export map', async () => {
+			// Deliberately denormalizes file paths and ensures named exports
+			// continue to work.
+			function hookedResolve() {
+				const resolvePlugin = resolve();
+				const oldResolve = resolvePlugin.resolveId;
+				resolvePlugin.resolveId = async function(source) {
+					const result = await oldResolve.apply(resolvePlugin, arguments);
+					if (source === 'external') {
+						result.id = result.id.replace(/\\/g, '/');
+						return result;
+					} else if (source.match(/custom-named-exports/)) {
+						result.id = result.id.replace(/\//g, '\\');
```

Comment:
I would assume this does not work on Linux/Mac as a backslash here is treated as a legal file name character and will not be changed by `normalize`, at least to my knowledge. As this problem is only relevant on Windows, maybe it is possible to write this test in a way that it trivially passes on Linux? I.e. do not replace `/` with `\`, only the other way around? Or is the test meaningless in such a situation?

Reply:
Alternatively, you could also import `path` into the test and use `sep`.

Reply:
I will push an update with the first approach later today (trivially passes on Linux).